test/e2e/arktos/multi_tenancy/test_suites/kubectl/test_attach.yaml

```diff
@@ -0,0 +1,56 @@
+#~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ Kubectl attach Tests ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+# This test suite verifies command "kubectl attach".
+#~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+######################################################
+# test variables
+######################################################
+Variables:
+  test_tenant: random_8
+  test_ns: random_8
+
+###########################################################################################################
+# test setup
+###########################################################################################################
+Tests:
+  - BeforeTestMessage: Starting test setup ...
+    Command: ${kubectl} create tenant ${test_tenant}
+    OutputShouldContain: 
+    - "\ntenant/${test_tenant} created\n"
+
+  - Command: ${kubectl} create namespace ${test_ns} --tenant ${test_tenant}
+    OutputShouldBe: "namespace/${test_ns} created\n"
+
+  - Command:  ${kubectl} apply -f ${test_data_dir}/sample-pod.yaml -n ${test_ns} --tenant ${test_tenant} 
+    OutputShouldBe: "pod/sample-nginx-pod created\n"
+
+############################################
+# test "kubectl attach"
+############################################
+
+# As the command "kubectl attach" does not return, we force it to return by prepending "timeout 2". 
+# As a result, we will have a non-zero exit code.
+# However, we can verify by checking the command output.
+
+  - BeforeTestMessage: "Verifying \"kubectl attach\" ..."
+    # need to wait the pod to be ready
+    BeforeTest: sleep 5
```

Comment:
curious. "need to wait the pod to be ready" do we have a way to periodically check pod status instead of "sleep 5"?

Reply:
here I'm assuming "sleep 5" is literally just sleep for 5 seconds

Reply:
So far sleep five seconds work fine. If it run into problems, I may turn to some more complicated methods, such as running "kubectl get pod" to check the pod status first.